ui/tests/acceptance/secrets/backend/aws/aws-configuration-test.js

```diff
@@ -190,30 +196,27 @@ module('Acceptance | aws | configuration', function (hooks) {
       await runCmd(`delete sys/mounts/${path}`);
     });
 
-    test('it should save lease AWS configuration', async function (assert) {
-      assert.expect(3);
+    test('it does not save lease AWS configuration if root configuration errored on save', async function (assert) {
+      assert.expect(1);
       const path = `aws-${this.uid}`;
       await enablePage.enable('aws', path);
 
       this.server.post(configUrl('aws', path), () => {
-        assert.false(
-          true,
-          'post request was made to config/root when no data was changed. test should fail.'
+        assert.ok('post request was made to config/root when no data was changed. test should fail.');
+        return overrideResponse(400, { errors: ['bad request!'] });
+      });
+      this.server.post(configUrl('aws-lease', path), () => {
+        assert.true(
+          false,
+          'post request was made to config/lease when the first config was not saved. test should fail.'
```

Comment:
I remember us talking about this, but why did you opt to use an assertion here instead of throwing an error if a request is made to this endpoint? 

If that's the case, can we change the assertion from "test should fail" (which isn't very clear) to "a request to this endpoint should NOT be made"